qcodes/tests/test_parameter.py

```diff
@@ -1594,14 +1595,47 @@ def _pp_setter(self, value):
     def _cp_setter(self, value):
         self._cp_counter += 1
 
+    def _cp_getter(self):
+        self._cp_get_counter += 1
+        return self.instrument['counting_parameter']._latest['value']
+
     def tearDown(self):
         self.instrument.close()
         del self.instrument
 
+    def test_set_to_none_when_current_value_is_none(self):
+        counting_parameter = self.instrument.counting_parameter
+        # Pre-conditions:
+        assert self._cp_counter == 0
+        assert self._cp_get_counter == 0
+        assert counting_parameter._latest["value"] is None
+        assert counting_parameter.get_latest.get_timestamp() is None
+
+        with counting_parameter.set_to(None):
+            # The value should not change
+            assert counting_parameter._latest["value"] is None
+            # The timestamp of the latest value should be still None
+            assert counting_parameter.get_latest.get_timestamp() is None
```

Comment:
I think the test may be wrong.
To me the name suggests a different test. It says that the current value is none, where the test sets up timestamp to be none, which means that the current state is actually unknown.
If it is unknown we have to get it on entering the context, so that we can restore it at the end.
Therefore we have to expect the timestamp to change. (See marked line here)

Reply:
got it. let me adjust the test....